modules/rest-util/test/blaze/middleware/fhir/resource_test.clj

```diff
@@ -190,21 +219,57 @@
   (testing "long attribute values are allowed (XML-wrapped Binary data)"
     (given @(resource-handler
              {:headers {"content-type" "application/fhir+xml"}
-              :body (input-stream (str "<Binary xmlns=\"http://hl7.org/fhir\"><data value=\"" (apply str (repeat (* 8 1024 1024) \a)) "\"/></Binary>"))})
+              :body (string-input-stream (str "<Binary xmlns=\"http://hl7.org/fhir\"><data value=\"" (apply str (repeat (* 8 1024 1024) \a)) "\"/></Binary>"))})
       fhir-spec/fhir-type := :fhir/Binary)))
 
+(deftest binary-test
+  (testing "when sending FHIR-wrapped Binary data"
+    (testing "both handlers should return the same for both wrappers (JSON and XML)"
+      (let [b64-encoded-binary-data "MTA1NjE0Cg=="
+            binary-resource-data (type/base64Binary b64-encoded-binary-data)
+            binary-resource-content-type "text/plain"]
+        (doseq [handler [resource-handler binary-resource-handler]
+                [fhir-content-type resource-string-representation]
+                [["application/fhir+json;charset=utf-8" (resource-as-json b64-encoded-binary-data binary-resource-content-type)]
+                 ["application/fhir+xml;charset=utf-8" (resource-as-xml b64-encoded-binary-data binary-resource-content-type)]]]
+          (let [closed? (atom false)]
+            (given @(handler
+                     {:headers {"content-type" fhir-content-type}
+                      :body (string-input-stream resource-string-representation closed?)})
+              {:fhir/type :fhir/Binary
+               :contentType (type/code binary-resource-content-type)
+               :data binary-resource-data})
```

Comment:
you can inline that binding, because it's only used once and should fit in one line